src/object.di

```diff
@@ -595,6 +595,82 @@ inout(V) get(K, V)(inout(V[K])* aa, K key, lazy inout(V) defaultValue)
     return (*aa).get(key, defaultValue);
 }
 
+private void _destructRecurse(S)(ref S s)
+    if (is(S == struct))
+{
+    import core.internal.traits : hasElaborateDestructor;
+
+    static if (__traits(hasMember, S, "__dtor"))
+        s.__dtor();
+
+    foreach_reverse (i, ref field; s.tupleof)
+    {
+        static if (hasElaborateDestructor!(typeof(field)))
+            _destructRecurse(field);
+    }
+}
+
+private void _destructRecurse(E, size_t n)(ref E[n] arr)
+{
+    import core.internal.traits : hasElaborateDestructor;
+
+    static if (hasElaborateDestructor!E)
+    {
+        foreach_reverse (ref elem; arr)
+            _destructRecurse(elem);
+    }
+}
+
+private string _genFieldPostblit(S)()
+{
+    import core.internal.traits : hasElaborateCopyConstructor;
+    import core.internal.traits : FieldNameTuple;
+
+    string code;
+    foreach(fieldName; FieldNameTuple!S)
+    {
+        static if(hasElaborateCopyConstructor!(typeof(__traits(getMember, S.init, fieldName))))
+        {
+            code ~= `
+                _postblitRecurse(s.` ~ fieldName ~ `);
+                scope(failure) _destructRecurse(s. ` ~ fieldName ~ `);
```

Comment:
I forget how "scope(failure)" deals with thrown `Errors`. I seem to remember that it catches them? Totally minor, but the correct implementation should not catch errors here.

The real reason I'm asking is I'm wondering how well the compiler optimizes away all these scope failure calls in case of nothrow functions (and in particular, if it can optimize those away at all, if it has to deal with thrown errors...)

So the question is: Did you test how efficient this code is? Are those catches optimized away? Maybe it warrants testing *if* the postblit can throw at all? Unfortunately, the test would require `functionAttributes`...